bentoml/_internal/io_descriptors/file.py

```diff
@@ -155,6 +165,41 @@ async def to_http_response(
             res = Response(body)
         return res
 
+    async def to_proto(self, obj: FileType) -> pb.File:
+        from bentoml.grpc.utils import mimetype_to_filetype_pb_map
+
+        if self._mime_type.startswith("multipart"):
+            raise UnprocessableEntity(
+                "'multipart' Content-Type is not yet supported for parsing files in gRPC. Use Multipart() instead."
+            ) from None
+
+        if isinstance(obj, bytes):
+            body = obj
+        else:
+            body = obj.read()
+
+        try:
+            kind = mimetype_to_filetype_pb_map()[self._mime_type]
+        except KeyError:
+            raise UnprocessableEntity(
+                f"{self._mime_type} doesn't have a corresponding File 'kind'"
+            ) from None
+
+        return pb.File(kind=kind, content=body)
+
+    if TYPE_CHECKING:
+
+        async def from_proto(
+            self,
+            field: pb.File | pb.Part | bytes,
+            *,
+            _use_internal_bytes_contents: bool = False,
+        ) -> FileLike[bytes]:
+            ...
+
+        async def from_http_request(self, request: Request) -> t.IO[bytes]:
+            ...
```

Comment:
Types from files without this will give error, which is a false positive and annoying
![Screenshot 2022-09-10 at 05 59 30](https://user-images.githubusercontent.com/29749331/189484463-783abbd5-8d6f-42cc-85a2-718ed247b5ea.png)

Reply:
Changing types for `__new__` is even worst, since we will have to overload `__new__` with the correct file type, I think that approach is even harder to maintain. I think comments should be fine.

Reply:
I'm not sure what you mean; that would be the correct type annotation and this is clearly a hack.

Reply:
That makes sense, let me update it.

Reply:
I think code that accommodates types checking for this one fine, and since the base IO alrady raise `NotImplementedError`.

Reply:
Huh, ok then.